src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2763,9 +2764,15 @@ private void EnsureDescriptorsInitialized()
 #endif
             if (m_eventData == null)
             {
+#if !ES_BUILD_STANDALONE
+                [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
```

Comment:
Why do you need to suppress this?

1. What method is marked with `RequiresUnreferencedCode`?
2. Since you have `DynamicallyAccessedMembers` on the class, the ILLinker should be able to see that you are calling `this.GetType()`, which is marked with `All`, and thus satisfies the `DynamicallyAccessedMembers` on `CreateManifestAndDescriptors`.

Reply:
As part of the trimmer work that we did related to object.gettype(), trimmer gets to see all members of all derived types and will treat all methods on EventSource as accessed via reflection. Some of these will have `RequiresUnreferencedCode` and it will warn from here. Hence the separation to a local method and suppression

Reply:
So the issue is:

1. We have methods on EventSource marked with `RequiresUnreferencedCode`.
2. We are annotating the whole EventSource class with `DynamicallyAccessedMembers.All`
3. The trimmer is seeing that we could be dynamically accessing method marked with `RequiresUnreferencedCode`, thus it is warning.

Do I have that correct?

If so, I think we need a better Justification or comment here explaining the situation. It isn't obvious from reading the current code.

Reply:
You are right, I'll try clarifying in the message